src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -35,7 +35,9 @@ namespace ErrorCodes
 
 DiskTransactionPtr DiskObjectStorage::createTransaction()
 {
-    return std::make_shared<FakeDiskTransaction>(*this);
+    if (use_fake_transaction)
+        return std::make_shared<FakeDiskTransaction>(*this);
+    return createObjectStorageTransaction();
```

Comment:
Why do we need this change?

Reply:
`FakeDiskTransaction` is a fake.
I'm testing the real one. In private we are using the real transactions. And this fix cover the issues there actually.